### DISCUSSION TITLE
Ubuntu 20.04 的 Pandas 不支持 indent json 输出

### DISCUSSION THREAD

**Discussion by heroxbd** [physics-data/faq#162] [Q&A] [answered]
黄赢业和吴律航两位发现作业模版中的 json 输出[1]，需要 Pandas 1.0 以上支持[2]。但是 Ubuntu 20.04 LTS 的 Pandas 版本不够[3]。

1. https://github.com/physics-data/tpl_medals2/blob/master/medals2.py#L25
2. https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_json.html
3. https://packages.ubuntu.com/focal/python3-pandas

**Comment by jiegec** [comment]
啊这，用 pip 装吧

**Comment by heroxbd** [comment, marked as answer]
临时解决方法是从 Ubuntu 20.10 安装 `python3-pandas`

在 `/etc/apt/sources.list` 中加入一行

```
deb https://mirrors.tuna.tsinghua.edu.cn/ubuntu/ groovy main restricted universe multiverse
```

然后以 root 运行

```
apt update
apt install python3-pandas
```

**Comment by mhliu0001** [comment]
我的系统是Ubuntu 20.04 LTS，但没有出现这个问题。或许应该用pip3装？

**Reply by heroxbd** [reply]
请给出

```python
import pandas as pd
print(pd.__file__)
```
的输出，如果是 `/usr/lib/python3/dist-packages/pandas/__init__.py` 代表是由系统的 APT 安装的，否则是其它方法安装的。

**Reply by mhliu0001** [reply]
`/home/mhliu/.local/lib/python3.8/site-packages/pandas/__init__.py`
我的python包都是用pip装的，pandas版本是1.3.2

**Reply by heroxbd** [reply]
所以你没有受到影响。
